flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/SourceCodeGenerator.java

```diff
@@ -422,4 +395,47 @@ private void generateInferredGraphqlClassField(Function function, StringBuilder
         stringBuilder.append(" ").append(function.name());
         stringBuilder.append(";");
     }
+
+    private String generateGraphqlClassResourceFunction(Function function) {
+        String docs = (function.description() != null && !function.description().isEmpty())
+                ? LS + "\t" + CommonUtils.convertToBalDocs(function.description())
+                : LS;
+
+        StringBuilder paramsBuilder = new StringBuilder();
+        for (int i = 0; i < function.parameters().size(); i++) {
+            Member param = function.parameters().get(i);
+            paramsBuilder
+                    .append(generateTypeDescriptor(param.type()))
+                    .append(" ")
+                    .append(param.name());
+            if (param.defaultValue() != null && !param.defaultValue().isEmpty()) {
+                paramsBuilder
+                        .append(" = ")
+                        .append(param.defaultValue());
+            }
+            if (i < function.parameters().size() - 1) {
```

Comment:
I would collect all the params to a single list of strings and join them with ", ". Instead of this if condition.